src/Microsoft.TestPlatform.Client/DesignMode/IDesignModeClient.cs

```diff
@@ -37,5 +32,18 @@ public interface IDesignModeClient : IDisposable
         /// Handles parent process exit
         /// </summary>
         void HandleParentProcessExit();
+
+        /// <summary>
+        /// Send the raw messages to IDE
+        /// </summary>
+        /// <param name="rawMessage"></param>
+        void SendRawMessage(string rawMessage);
+
+        /// <summary>
+        /// Send the test session messages to IDE
+        /// </summary>
+        /// /// <param name="level"></param>
+        /// <param name="message"></param>
+        void SendTestMessage(TestMessageLevel level, string message);
```

Comment:
We would also need a change in Translation Layer, so that it knows a new message is coming in, & it needs to handle it?

Reply:
It can already handle these messages. It is aware of TestMessages.
TestMessages flow as Raw messages since they are already serialized.